project/TestParallelization.scala

```diff
@@ -107,25 +116,38 @@ object TestParallelization {
     def testGroups: List[Tests.Group]
   }
 
-  class SimpleHashStrategy private(groups: Map[Int, Tests.Group]) extends GroupingStrategy {
+  class SimpleHashStrategy private(
+      groups: Map[Int, Tests.Group],
+      shard: Option[Int]) extends GroupingStrategy {
 
     lazy val testGroups = groups.values.toList
     val groupCount = groups.size
 
     override def add(testDefinition: TestDefinition): GroupingStrategy = {
       val groupIdx = math.abs(testDefinition.name.hashCode % groupCount)
+      if (shard.isDefined) {
```

Comment:
add some comments here to explain this please. After a few minutes I still don't 100% understand the logic

Reply:
I updated with a comment, please let me know if it still is unclear, I think that's important to make sure everyone's confident on this sharding logic!